tensorflow/stream_executor/rocm/rocm_blas.cc

```diff
@@ -416,6 +416,105 @@ bool ROCMBlas::DoBlasInternalImpl(FuncT rocblas_func, Stream *stream,
   return ret == rocblas_status_success;
 }
 
+
+template <class T>
+void SplitComplex(T* as, const T* a, int n, void* stream, bool conjugate);
+template <class T>
+void CombineComplex(T* a, const T* as, int n, void* stream);
+template <class T, int N>
+void AddComplex(T* as, const T* a, int n, void* stream);
+
+template <typename FuncT, typename T, typename... Args> 
+port::Status ROCMBlas::DoBlasComplex(ScratchAllocator* scratch_allocator,
+    Stream* stream,
+    const T* a, const T* b, T* c,
+    uint64_t na, uint64_t nb, uint64_t nc, 
+    std::complex<T> alpha, std::complex<T> beta,
+    FuncT rocblas_func,
+    int conj_op,
+    Args... args)
+{
+    if(alpha.imag()!=0.0 || beta.imag()!=0.0)
+      return port::Status(
+          port::error::INTERNAL, "ROCMBlas only supports real-valued alpha & beta");
+    auto hs = AsGpuStreamValue(stream);
+    auto sz_alloc = na * sizeof(T) * 2 + nb *sizeof(T)*2 + nc*sizeof(T)*4;
```

Comment:
small comment here explaining how the scratch memory is used, will be helpful in understanding the size calculations.